Dtspb 3394 default email notification for grant issue to yes

AbdullahShahHMCTS opened the pull request:
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPB-3394


### Change description ###
The radio button is now selected as yes for the notificaiton of the grant issue.


**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

AbdullahShahHMCTS committed:
accept master changes

AbdullahShahHMCTS committed:
add test for defaulting response

AbdullahShahHMCTS committed:
refactored test.
now the option is true, even when the email is blank

AbdullahShahHMCTS committed:
refactored test.
now the option is true, even when the email is blank

sanjayparekh-hmcts commented on src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseData.java:
```diff
@@ -249,7 +249,8 @@ public class CaseData extends CaseDataParent {
 
-    private final String boEmailGrantIssuedNotification;
+    @Builder.Default
+    private final String boEmailGrantIssuedNotification = YES;
```
I dont think you can set it this way, because there are situations where there is no email address to send the notification... and this would break that

AbdullahShahHMCTS replied:
Have removed this and added logic in transformer

AbdullahShahHMCTS committed:
DTSPB-3394
reverted the default answer from yes
updated tests

AbdullahShahHMCTS committed:
DTSPB-3394
reverted test

AbdullahShahHMCTS committed:
DTSPB-3394
reverted test

AbdullahShahHMCTS committed:
DTSPB-3394
updated the logic, to check for whether its sol or pa application

AbdullahShahHMCTS committed:
DTSPB-3394
updated the logic, to check for null casedata types

AbdullahShahHMCTS committed:
Merge branch 'master' into DTSPB-3394_Default_Email_Notif_to_Yes

AbdullahShahHMCTS committed:
DTSPB-3394
make jenkins run code changes

AbdullahShahHMCTS committed:
DTSPB-3394
make jenkins run code changes

AbdullahShahHMCTS committed:
Merge branch 'master' into DTSPB-3394_Default_Email_Notif_to_Yes

AbdullahShahHMCTS committed:
DTSPB-3394
checkstyle

AbdullahShahHMCTS committed:
DTSPB-3394
checkstyle

AbdullahShahHMCTS committed:
DTSPB-3394
update tests to now follow new logic, that if the solsemail is set and the application type is sols, default notifs to yes

AbdullahShahHMCTS committed:
Merge branch 'master' into DTSPB-3394_Default_Email_Notif_to_Yes

sanjayparekh-hmcts commented on src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java:
```diff
@@ -1365,32 +1365,34 @@ private void updateCaseBuilder(CaseData caseData, ResponseCaseDataBuilder<?, ?>
-                    .boEmailRequestInfoNotification(ANSWER_NO)
-                    .boEmailGrantIssuedNotification(ANSWER_NO)
-                    .boEmailGrantReissuedNotification(ANSWER_NO);
+        if (caseData.getApplicationType() != null && caseData.getApplicationType().equals(SOLICITOR)) {
```
we should always put this kinda condition in the opposite way:
SOLICITOR.equals(caseData.getApplicationType())
then you dont need a null check

and you could just do:
String answer = isSolsEmailSet(caseData) ? ANSWER_YES : ANSWER_NO;
builder..boEmailDocsReceivedNotification(answer)
                        .boEmailRequestInfoNotification(answer)
                        .boEmailGrantIssuedNotification(answer)
                        .boEmailGrantReissuedNotification(answer);

same for PERSONAL block below

AbdullahShahHMCTS committed:
DTSPB-3394
updated code to be more concise following feedback

AbdullahShahHMCTS committed:
DTSPB-3394
checkstyle issues

AbdullahShahHMCTS committed:
DTSPB-3394
checkstyle issues

AbdullahShahHMCTS committed:
Merge branch 'master' into DTSPB-3394_Default_Email_Notif_to_Yes

FeliTam committed:
Merge branch 'master' into DTSPB-3394_Default_Email_Notif_to_Yes
